homeassistant/components/tibber/services.py

```diff
@@ -0,0 +1,94 @@
+"""Services for Tibber integration."""
+
+from __future__ import annotations
+
+import datetime as dt
+from datetime import date, datetime
+from functools import partial
+from typing import Final
+
+import voluptuous as vol
+
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+    callback,
+)
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .sensor import TibberDataCoordinator
+
+PRICE_SERVICE_NAME = "get_prices"
+ATTR_START: Final = "start"
+ATTR_END: Final = "end"
+
+SERVICE_SCHEMA: Final = vol.Schema(
+    {
+        vol.Optional(ATTR_START): str,
+        vol.Optional(ATTR_END): str,
+    }
+)
+
+
+async def __get_prices(call: ServiceCall, *, hass: HomeAssistant) -> ServiceResponse:
+    tibber_connection = hass.data[DOMAIN]
+    coordinator = TibberDataCoordinator(hass, tibber_connection)
+
+    for home in tibber_connection.get_homes(only_active=False):
+        tibber_home = home
+
+    start = __get_date(call.data.get(ATTR_START), "start")
+    end = __get_date(call.data.get(ATTR_END), "end")
+
+    if start >= end:
+        end = start + dt.timedelta(days=1)
```

Comment:
This block seems like the kind of logic that's likely to be poorly covered, likely unused, and possibly harbor a bug. And indeed, the test cases don't seem to cover this code. I suspect it passes without this block. I recommend to either add test cases to illustrate why this logic is necessary or remove it.

Reply:
This is to ensure when a user sets a start date after today(tomorrow) that it also looks for data to the end of that day. By default, only one day is shown unless you change the parameters. Without this statement, no data is added if tomorrow is selected as a start.

I am not yet good at writing full-coverage tests so still need to add them

Reply:
I'm not sure what the practice is for similar service calls. To me it would make most sense to raise an error if start >= end

Reply:
The Easyenergy integration and its' virtual counterparts add an `unknown error` because there is no error raised from the code base.

I thought to implement it this way to not throw a specific or `unknown` error if the user only specifies the `start_time` to the date of tomorrow. This means you can set the `start_time` to tomorrow and leave the `end_time` empty to get the full data for tomorrow.

In this way, it will always return data (if available, else an empty list) to the end of the day that is used as `start_time` unless an earlier `end_time` is specified.

Reply:
Added this case to the tests. Imho this approach is the most user-friendly (no need to set an end date if the start date is set to tomorrow's date).